WalletWasabi/WabiSabi/Client/CoinJoinManager.cs

```diff
@@ -90,24 +90,31 @@ protected override async Task ExecuteAsync(CancellationToken stoppingToken)
 					trackedWallets.Remove(finishedCoinJoin.Wallet.WalletName);
 					finishedCoinJoin.CancellationTokenSource.Dispose();
 
-					var finishedCoinJoinTask = finishedCoinJoin.CoinJoinTask;
-					if (finishedCoinJoinTask.IsCompletedSuccessfully)
+					try
 					{
-						Logger.LogInfo("Coinjoin client finished successfully!");
-					}
-					else if (finishedCoinJoinTask.IsFaulted)
-					{
-						if (finishedCoinJoinTask.Exception?.InnerException is InvalidOperationException ioe)
+						var success = await finishedCoinJoin.CoinJoinTask.ConfigureAwait(false);
+						if (success)
+						{
+							Logger.LogInfo("Coinjoin client finished successfully!");
+						}
+						else
 						{
-							await Task.Delay(TimeSpan.FromSeconds(5), stoppingToken).ConfigureAwait(false);
-							continue;
+							Logger.LogInfo("Coinjoin client finished with error. Transaction not broadcasted.");
```

Comment:
It seems to me that the wallet name in the log may be useful.

Reply:
Agree. Done